packages/search-ui/src/DebounceManager.js

```diff
@@ -1,30 +1,58 @@
 import debounceFn from "debounce-fn";
 
-export default class DebounceManager {
+class DebounceManager {
   debounceCache = {};
 
-  /*
-  The purpose of this is to:
-  Dynamically debounce and cache a debounced version of a function at the time of calling that function. This avoids
-  managing debounced version of functions locally.
-
-  Assumption:
-  Functions are debounced on a combination of unique function and wait times. So debouncing won't work on
-  subsequent calls with different wait times or different functions. That also means that the debounce manager
-  can be used for different functions in parallel, and keep the two functions debounced separately.
-
-  */
-  runWithDebounce(wait, fn, ...parameters) {
+  /**
+   * Dynamically debounce and cache a debounced version of a function at the time of calling that function. This avoids
+   * managing debounced version of functions locally.
+   *
+   * In other words, debounce usually works by debouncing based on
+   * referential identity of a function. This works by comparing provided function names.
+   *
+   * This also has the ability to short-circuit a debounce all-together, if no wait
+   * time is provided.
+   *
+   * Assumption:
+   * Functions are debounced on a combination of unique function name and wait times. So debouncing won't work on
+   * subsequent calls with different wait times or different functions. That also means that the debounce manager
+   * can be used for different functions in parallel, and keep the two functions debounced separately.
+   *
+   *
+   * @param {number} wait Milliseconds to debounce. Executes immediately if falsey.
+   * @param {function} fn Function to debounce
+   * @param {function} functionName Name of function to debounce, used to create a unique key
+   * @param  {...any} parameters Parameters to pass to function
+   */
+  runWithDebounce(wait, functionName, fn, ...parameters) {
     if (!wait) {
       return fn(...parameters);
     }
 
-    const key = fn.toString() + wait.toString();
+    const key = functionName + wait.toString();
     let debounced = this.debounceCache[key];
     if (!debounced) {
       this.debounceCache[key] = debounceFn(fn, { wait });
       debounced = this.debounceCache[key];
     }
     debounced(...parameters);
   }
+
+  cancelByName(functionName) {
+    Object.entries(this.debounceCache)
+      .filter(([cachedKey]) => cachedKey.startsWith(functionName))
+      // eslint-disable-next-line no-unused-vars
+      .forEach(([_, cachedValue]) => cachedValue.cancel());
+  }
 }
+/**
+ * Perform a standard debounce
+ *
+ * @param {number} wait Milliseconds to debounce. Executes immediately if falsey.
+ * @param {function} fn Function to debounce
+ */
+DebounceManager.debounce = (wait, fn) => {
```

Comment:
Just curious - why put this outside the class instead of keeping it inside? e.g.

```js
class DebounceManager {
  ...

  /**
   * ...
   */
  debounce(wait, fn) {
    return debounceFn(fn, { wait });
  }
}
```

Reply:
I wanted to make it a static method, so I didn't have to instantiate a new instance to use it.

I.e., so I could do this:

```
DebounceManager.debounce(...
```

Instead of:
```
const manager = new DebounceManager();
manager.debounce()
```

Reply:
Ahhh okay. That gives me a tiny amount of heebly jeeblies but I don't hate it enough to want to change it 👻

Reply:
Lol, static methods are a common thing though right?